### PR TITLE
Add toplevel and zoo system of Owl library

### PR DESCRIPTION
**Pull request by ryanrhymes**
This PR contains the Toplevel module and Zoo System for Owl library.

### CODE REVIEW

**Comment by mseri** [comment]
Somwthing wrong with the C dependencies or some linking flags?

```
+ jbuilder "build" "-p" "owl_zoo" "-j" "2" (CWD=/home/opam/.opam/4.05.0/build/owl_zoo.0.3.0)
[ERROR] The compilation of owl_zoo failed at "jbuilder build -p owl_zoo -j 2".
-     ocamldep bin/zoo.depends.ocamldep-output
-     ocamldep src/zoo/owl_zoo.depends.ocamldep-output
-       ocamlc src/zoo/owl_zoo_cmd.{cmi,cmo,cmt}
-       ocamlc src/zoo/owl_zoo_dir.{cmi,cmo,cmt}
-       ocamlc src/zoo/owl_zoo.cma
-       ocamlc bin/zoo.{cmi,cmo,cmt}
-       ocamlc bin/zoo.bc (exit 2)
- (cd _build/default && /home/opam/.opam/4.05.0/bin/ocamlc.opt -w -40 -safe-string -g -o bin/zoo.bc -linkall -I /home/opam/.opam/4.05.0/lib/bytes -I /home/opam/.opam/4.05.0/lib/ctypes -I /home/opam/.opam/4.05.0/lib/dolog -I /home/opam/.opam/4.05.0/lib/eigen -I /home/opam/.opam/4.05.0/lib/gsl -I /home/opam/.opam/4.05.0/lib/integers -I /home/opam/.opam/4.05.0/lib/ocaml -I /home/opam/.opam/4.05.0/lib/ocaml-compiler-libs/toplevel -I /home/opam/.opam/4.05.0/lib/ocaml/compiler-libs -I /home/opam/.opam/4.05.0/lib/owl -I /home/opam/.opam/4.05.0/lib/plplot -I src/zoo /home/opam/.opam/4.05.0/lib/ocaml/unix.cma /home/opam/.opam/4.05.0/lib/ocaml/bigarray.cma /home/opam/.opam/4.05.0/lib/integers/integers.cma /home/opam/.opam/4.05.0/lib/ctypes/ctypes.cma /home/opam/.opam/4.05.0/lib/ocaml/str.cma /home/opam/.opam/4.05.0/lib/ctypes/cstubs.cma /home/opam/.opam/4.05.0/lib/dolog/dolog.cma /home/opam/.opam/4.05.0/lib/eigen/eigen.cma /home/opam/.opam/4.05.0/lib/gsl/gsl.cma /home/opam/.opam/4.05.0/lib/plplot/plplot.cma /home/opam/.opam/4.05.0/lib/owl/owl.cma /home/opam/.opam/4.05.0/lib/ocaml/compiler-libs/ocamlcommon.cma /home/opam/.opam/4.05.0/lib/ocaml/compiler-libs/ocamlbytecomp.cma /home/opam/.opam/4.05.0/lib/ocaml/compiler-libs/ocamltoplevel.cma /home/opam/.opam/4.05.0/lib/ocaml-compiler-libs/toplevel/ocaml_toplevel.cma src/zoo/owl_zoo.cma bin/zoo.cmo)
- File "_none_", line 1:
- Error: Error on dynamically loaded library: /home/opam/.opam/4.05.0/lib/stublibs/dllowl_stubs.so: /home/opam/.opam/4.05.0/lib/stublibs/dllowl_stubs.so: undefined symbol: LAPACKE_dtpmqrt
```

File: packages/owl_zoo/owl_zoo.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
+opam-version: "1.2"
+maintainer: "Liang Wang (ryanrhymes@gmail.com)"
+authors: [ "Liang Wang (ryanrhymes@gmail.com)" ]
+license: "MIT"
+homepage: "https://github.com/ryanrhymes/owl"
+dev-repo: "git+https://github.com/ryanrhymes/owl.git"
+bug-reports: "https://github.com/ryanrhymes/owl/issues"
+doc: "http://www.cl.cam.ac.uk/~lw525/owl/"
+
+version: "dev"
```
Comment (jpdeplaix) [inline packages/owl_zoo/owl_zoo.0.3.0/opam]: This line should be removed (or changed)

File: packages/owl_top/owl_top.0.3.0/opam [hunk 2]
```diff
@@ -0,0 +1,24 @@
+opam-version: "1.2"
+maintainer: "Liang Wang (ryanrhymes@gmail.com)"
+authors: [ "Liang Wang (ryanrhymes@gmail.com)" ]
+license: "MIT"
+homepage: "https://github.com/ryanrhymes/owl"
+dev-repo: "git+https://github.com/ryanrhymes/owl.git"
+bug-reports: "https://github.com/ryanrhymes/owl/issues"
+doc: "http://www.cl.cam.ac.uk/~lw525/owl/"
+
+version: "dev"
```
Comment (jpdeplaix) [inline packages/owl_top/owl_top.0.3.0/opam]: Same here.

**Comment by mseri** [comment]
I'll see if I understand what is wrong between this and next week

File: packages/owl_zoo/owl_zoo.0.3.0/opam [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+build: [
+  ["jbuilder" "subst" "-n" name] {pinned}
+  ["jbuilder" "build" "-p" name "-j" jobs]
+]
+build-test: [["jbuilder" "runtest" "-p" name "-j" jobs]]
+
+depends: [
+  "owl"
```
Comment (mseri) [inline packages/owl_zoo/owl_zoo.0.3.0/opam]: You need to require `"owl" { >= "0.3.0" }` to fix some of the issues

File: packages/owl_top/owl_top.0.3.0/opam [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+build: [
+  ["jbuilder" "subst" "-n" name] {pinned}
+  ["jbuilder" "build" "-p" name "-j" jobs]
+]
+build-test: [["jbuilder" "runtest" "-p" name "-j" jobs]]
+
+depends: [
+  "owl"
+  "owl_zoo"
```
Comment (mseri) [inline packages/owl_top/owl_top.0.3.0/opam]: Similarly here;
```  
  "owl" { >= "0.3.0" }
  "owl_zoo" { >= "0.3.0" }
```

**Comment by mseri** [comment]
I also see failures with missing `-lgfortran`

**Comment by ryanrhymes** [comment]
@mseri @jpdeplaix thanks for help, I will have a look at these issues later, back to work slowly now :)

**Pull request closed without merging**